File: LeetCode/Medium/0735-asteroid-collision/0735-asteroid-collision.cpp
class Solution {
public:
    vector<int> asteroidCollision(vector<int>& asteroids) {
        int len = asteroids.size();
        for(int i = 0; i < len; i++){
            for(int j = 1; j < asteroids.size(); j++){
                if(asteroids[j-1] > 0 && asteroids[j] < 0){
                    if(asteroids[j-1] > -1 * asteroids[j]){
                        asteroids.erase(asteroids.begin() + j);
                    }
                    else if(asteroids[j-1] < -1 * asteroids[j]){
                        asteroids.erase(asteroids.begin() + j - 1);
                    }
                    else if(asteroids[j-1] == -1 * asteroids[j]){
                        asteroids.erase(asteroids.begin() + j - 1, asteroids.begin() + j + 1);
                    }
                }
            }
        }
        return asteroids;
    }
};

File: LeetCode/Medium/0735-asteroid-collision/0735-asteroid-collision.java
class Solution {
    public int[] asteroidCollision(int[] asteroids) {
        Stack<Integer> s = new Stack<>();
        for (int i = 0; i < asteroids.length; i++) {
            if (s.size() > 0 && asteroids[i] < 0 && s.peek() > 0) {
                if (asteroids[i] * -1 == s.peek()) s.pop();
                else if (asteroids[i] * -1 > s.peek()) {
                    s.pop();
                    i--;
                }
            } else s.add(asteroids[i]);
        }
        return s.stream().mapToInt(i -> i).toArray();
    }
}